### PR TITLE
Tidied up dependencies and made the juce_audio_processors optional.

### PR DESCRIPTION
**Pull request by jrlanglois**
See #148

### CODE REVIEW

File: blueprint/blueprint.h [hunk 1]
```diff
@@ -11,7 +11,7 @@
     name:               Blueprint
     description:        A React.js render backend targeting JUCE Components.
     minimumCppStandard: 17
-    dependencies:       juce_core, juce_graphics, juce_gui_basics
+    dependencies:       juce_gui_basics
```
Comment (jrlanglois) [inline blueprint/blueprint.h:14]: The Projucer is smart enough to figure this out...

File: blueprint/blueprint.h [hunk 2]
```diff
@@ -20,11 +20,11 @@
 
 #define BLUEPRINT_H_INCLUDED
 
-
-#include <juce_core/juce_core.h>
-#include <juce_graphics/juce_graphics.h>
 #include <juce_gui_basics/juce_gui_basics.h>
```
Comment (jrlanglois) [inline blueprint/blueprint.h:23]: This module recursively adds the headers that I removed above.

**Comment by nick-thompson** [comment]
Ah! This makes a ton of sense @jrlanglois, thanks very much!

I'm quite ready to merge, just have one last question: will this break existing projects? I don't think so, right? Because those existing projects will have been forced already to add all of these juce module dependencies. When this merges, those existing projects wont' break, but will now have the opportunity to remove their dep on juce_audio_processors. Is that right?

**Comment by jrlanglois** [comment]
It shouldn't have any effect on other projects!

> When this merges, those existing projects wont' break, but will now have the opportunity to remove their dep on juce_audio_processors. Is that right?

That's precisely the it: assuming a non-audio app is in the works, the user can simply pull that module out of their project now.

**Comment by nick-thompson** [comment]
Excellent, then I'll go ahead and merge! Thanks very much @jrlanglois

**Pull request merged**